src/puppetlabs/comidi.clj

```diff
@@ -21,17 +21,24 @@
 (def Zipper
   (schema/pred ks/zipper?))
 
-(def http-methods
-  #{:any :get :post :put :delete :head :options})
-
 (def RequestMethod
   (schema/enum :any :get :post :put :delete :head :options))
 
+; Derived from bidi-schema PatternSegment
 (def RegexPatternSegment
-  [(schema/one Pattern "regex") (schema/one schema/Keyword "variable")])
+  (schema/pair schema/Regex "qual" schema/Keyword "id"))
+
+; Derived from bidi-schema by flattening a vector of Patterns
+(def RouteInfoPathSegment
+  (schema/either bidi-schema/Path
+                 bidi-schema/PatternSegment
+                 schema/Bool))
```

Comment:
OK, so the attached horror story of a picture compares ```s/either``` with ```s/cond``` in this particular context. I feel like ```s/either``` isn't really worse and makes my intention a bit more clear. Not sure how that works for you.

![screen shot 2015-09-04 at 15 09 44](https://cloud.githubusercontent.com/assets/3063979/9685632/7593989e-5317-11e5-8783-80d9472a70af.png)

Here's the code using s/ cond:
```clojure
(def RouteInfoPathSegment
  (schema/conditional
   #(nil? (schema/check bidi-schema/Path %)) schema/Any
   #(nil? (schema/check bidi-schema/PatternSegment %)) schema/Any
   :else schema/Bool))
```